src/screens/ChooseWallet.js

```diff
@@ -0,0 +1,215 @@
+/**
+ * Copyright (c) Hathor Labs and its affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import React from 'react';
+import { t } from 'ttag';
+
+import wallet from '../utils/wallet';
+import helpers from '../utils/helpers';
+import hathorLib from '@hathor/wallet-lib';
+import logo from '../assets/images/hathor-logo.png';
+import InitialImages from '../components/InitialImages';
+import ModalResetAllData from '../components/ModalResetAllData';
+import colors from '../index.scss';
+import ReactLoading from 'react-loading';
+import $ from 'jquery';
+
+/**
+ * Show list of wallets for user to choose from and ask for PIN.
+ *
+ * @memberof Screens
+ */
+class ChooseWallet extends React.Component {
+  constructor(props) {
+    super(props);
+
+    // update storage in case it did not support multiple wallets (wallet upgrade)
+    const wallets = hathorLib.storage.store.getListOfWallets();
+    if (!wallets || Object.keys(wallets).length === 0){
+      const name = 'default';
+      const prefix = wallet.walletNameToPrefix(name);
+      hathorLib.storage.store.upgradeStorage(prefix, name);
+    }
+
+    this.state = {
+      selectedWallet: null,
+      errorMessage: null,
+      loading: false,
+      wallets: hathorLib.storage.store.getListOfWallets(),
+    }
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    if (prevState.selectedWallet !== this.state.selectedWallet) {
+      this.refs.pin.focus();
+    }
+  }
+
+  /**
+   * Called when user clicks on a wallet name.
+   *
+   * @param {Object} e Event for the click
+   * @param {String} prefix Prefix of the chosen wallet
+   */
+  goToWallet = (e, prefix) => {
+    e.preventDefault();
+
+    if (this.state.loading) return;
+    
+    this.setState({ selectedWallet: prefix, errorMessage: null });
+  }
+
+  /**
+   * Called when user clicks on "Add Wallet" button.
+   */
+  addWallet = () => {
+    if (this.state.loading) return;
+    wallet.setWalletPrefix(null);
```

Comment:
If the user is in the `ChooseWallet` screen, the prefix should be `null` and the wallet data should've been cleaned already, right?

(The same for the hardware wallet and the code there is really similar to this one, we should reuse part of it)

Edit: forget about the prefix, you need to select it if the user clicks to remove a wallet but the data should be cleaned already, I guess.